compiler/rustc_middle/src/ty/util.rs

```diff
@@ -132,9 +132,9 @@ impl<'tcx> TyCtxt<'tcx> {
     /// context it's calculated within. This is used by the `type_id` intrinsic.
     pub fn type_id_hash(self, ty: Ty<'tcx>) -> Hash128 {
         // We want the type_id be independent of the types free regions, so we
-        // erase them. The erase_regions() call will also anonymize bound
+        // erase them. The erase_and_anonymize_regions() call will also anonymize bound
         // regions, which is desirable too.
```

Comment:
while you're here please rewrite this comment, the vibe is off.

we don't want this. we need this for soundness.

> We don't have region information, so we erase all free regions. Equal types
> also need to have the same `TypeId`, so we must anonymize all bound regions
> as well.